Cabal/Distribution/Fields/Pretty.hs

```diff
@@ -34,6 +34,7 @@ import qualified Text.PrettyPrint as PP
 
 data PrettyField ann
     = PrettyField ann FieldName PP.Doc
+    | PrettyFieldCommentedOut ann FieldName
```

Comment:
This doesn't look right.

I think we should rather extend `ann` to carry notion whether thing is "commented" out. E.g. whole `PrettySection` could be commented out - that sounds as "useful".

I.e. maybe there should be `showFields'` variant which also takes `(ann -> Bool)` argument and comments out stuff if the annotation says so.

Reply:
Related to this. Could you add a golden file which generates commented out fields before this PR.

Reply:
It's not as invasive change as adding a new constructor.  See https://github.com/haskell/cabal/compare/master...phadej:pretty-post for a PoC

```haskell
> putStr $ showFields' fst snd 2 [PrettyField (["-- comm comm"], id) "foo" (PP.text "value"), PrettyField ([], map ("-- " ++)) "bar" (PP.text "value2") ]
-- comm comm
foo: value
-- bar: value2
```

Reply:
Makes sense, updated the code and introduced a new `FieldAnnotation` type in `FileCreators.hs` to hold the comments and the decision on whether or not to comment out the field. No diffs in goldens (as expected).